[TLVB-23] 엔티티 클래스 구현

dia312 opened the pull request:
## 👩‍💻 요구 사항과 구현 내용
- Entity 클래스를 구현
- @Column 에 ERD에 있는 제약조건 설정
- 필요한 생성자에 @Builder 적용
- 패키지 구조는 크게 controller, service, domain 으로 분리하였고, domain 안에는 entity를 기준으로 entity 클래스와 repository 클래스 그리고 vo 클래스를 넣기로 하였습니다. 

## ✅ 피드백 반영사항


## 🤔 PR 포인트 & 궁금한 점
- email 과 같은 VO 객체로 타입 변경이 필요한 부분이 뭐가 있을지 논의 후 추가 적용합시당.

## 관련 이슈 
TLVB-23

dia312 committed:
feat: 엔티티 클래스 작성

dia312 committed:
feat: User 엔티티 추가

nasaoreo committed:
chore: lombok dependency 추가

ksy90101 commented on src/main/java/kdt/prgrms/kazedon/everevent/domain/like/Like.java:
```diff
@@ -0,0 +1,36 @@
+import lombok.NoArgsConstructor;
+
+@Entity
+@Table(name = "like")
```
SQL 예약어랑 동일해서 문제가 생길거 같은데요?

dia312 replied:
맞네요,, 팀원들이랑 얘기해서 다른 용어로 수정하겟슴다!

ksy90101 commented on src/main/java/kdt/prgrms/kazedon/everevent/domain/user/User.java:
```diff
@@ -0,0 +1,45 @@
+
+  @Id
+  @Column(name = "id")
+  @GeneratedValue(strategy = GenerationType.AUTO)
```
ID를 Auto로 한 이유가 있나요?

dia312 replied:
테스트 db랑 배포용 db를 h2랑 mysql로 2개를 사용해서 auto로 설정했습니당..!

Heejeong01110 reviewed (approved):
말씀하신 부분 적용 되면 머지해도 될것 같습니당!!

dia312 committed:
chore: 코드리뷰 피드백 적용
- @Column name 설정 삭제
- Review Entity pictureUrl nullable 설정 변경
- @ID 생성전략 변경
- Authority Entity authorityName 필드 추가
